src/main/scala/com/fulcrumgenomics/fasta/SequenceDictionary.scala

```diff
@@ -0,0 +1,206 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2020 Fulcrum Genomics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.fasta
+
+import com.fulcrumgenomics.FgBioDef
+import com.fulcrumgenomics.FgBioDef._
+import enumeratum.EnumEntry
+import htsjdk.samtools.{SAMSequenceDictionary, SAMSequenceRecord}
+import scala.collection.mutable
+import SequenceMetadata.Keys
+
+object SequenceMetadata {
+  /** Keys for standard attributes in [[SequenceMetadata]] */
+  object Keys {
+    val Aliases              : String = "AN"
+    val AlternateLocus       : String = "AH"
+    val Assembly             : String = SAMSequenceRecord.ASSEMBLY_TAG
+    val Description          : String = SAMSequenceRecord.DESCRIPTION_TAG
+    private[fasta] val Length: String = SAMSequenceRecord.SEQUENCE_LENGTH_TAG
+    val Md5                  : String = SAMSequenceRecord.MD5_TAG
+    private[fasta] val Name  : String = SAMSequenceRecord.SEQUENCE_NAME_TAG
+    val Species              : String = SAMSequenceRecord.SPECIES_TAG
+    val Topology             : String = "TP"
+    val Uri                  : String = SAMSequenceRecord.URI_TAG
+  }
+}
+
+/** Stores information about a single Sequence (ex. chromosome, contig)
+  *
+  * @param name the primary name of the sequence
+  * @param length the length of the sequence, or zero if unknown
+  * @param index the index in the sequence dictionary
+  * @param attributes attributes of this sequence
+  */
+case class SequenceMetadata(name: String,
+                            length: Int,
+                            index: Int = 0,
+                            attributes: Map[String, String] = Map.empty) {
+  allNames.foreach { name => SAMSequenceRecord.validateSequenceName(name) }
+  require(length >= 0, s"Length must be >= 0 for '$name'")
+  require(attributes.keys.forall(_ != Keys.Name),
```

Comment:
Why not ` require(!attributes.contains(Keys.Name), "...") and the same for `Length` to avoid the iteration?